hello-world/tutorial/index.md

```diff
@@ -0,0 +1,500 @@
+The content of the `sender` of the message is automatically included in `SendIbcPost` message. The sender is verified as the signer of the message, so you can add the `msg.Sender` as the creator to the new packet before it is sent over IBC.
+
+```go
+// planet/x/blog/keeper/msg_server_ibcPost.go
+// Construct the packet
+	var packet types.IbcPostPacketData
+
+	packet.Title = msg.Title
+	packet.Content = msg.Content
+	packet.Creator = msg.Sender // < ---
+	
+	// Transmit the packet
+	err := k.TransmitIbcPostPacket(
+		ctx,
+		packet,
+		msg.Port,
+		msg.ChannelID,
+		clienttypes.ZeroHeight(),
+		msg.TimeoutTimestamp,
+	)
+```
+
+## Receive the Post
+
+The methods for primary transaction logic are in the `planet/x/blog/keeper/ibcPost.go` file. Use these methods to manage IBC packets:
+
+- `TransmitIbcPostPacket` is called manually to send the packet over IBC. This method also defines the logic before the packet is sent over IBC to another blockchain app.
+- `OnRecvIbcPostPacket` hook is automatically called when a packet is received on the chain. This method defines the packet reception logic.
+- `OnAcknowledgementIbcPostPacket` hook is called when a sent packet is acknowledged on the source chain. This method defines the logic when the packet has been received.
+- `OnTimeoutIbcPostPacket` hook is called when a sent packet times out. This method defines the logic when the packet is not received on the target chain
+
+You must modify the source code to add the logic inside those functions so that the data tables are modified accordingly.
+
+On reception of the message, we create a new post with the title and the content on the receiving chain.
+
+To identify the blockchain app that a message is originating from and who created the message, use an identifier in the following format:
+
+`<portID>-<channelID>-<creatorAddress>`
+
+Finally, the AppendPost function that is generated by Starport returns the ID of the new appended post. You can return this value to the source chain through acknowledgment.
+
+Append the type instance as `PostID` on receiving the packet:
+
+- the context `ctx` is an [immutable data structure](https://docs.cosmos.network/master/core/context.html#go-context-package) that has header data from the transaction. [See how the context is initiated](https://github.com/cosmos/cosmos-sdk/blob/master/types/context.go#L71)
+- the identifier format that you defined earlier
+- the `title` is the Title of the blog post
+- the `content` is the Content of the blog post
+
+```go
+// planet/x/blog/keeper/ibcPost.go
+// OnRecvIbcPostPacket processes packet reception
+func (k Keeper) OnRecvIbcPostPacket(ctx sdk.Context, packet channeltypes.Packet, data types.IbcPostPacketData) (packetAck types.IbcPostPacketAck, err error) {
+	// validate packet data upon receiving
+	if err := data.ValidateBasic(); err != nil {
+		return packetAck, err
+	}
+
+	packetAck.PostID = k.AppendPost(
```

Comment:
@Tosch110 [IMPORTANT]
The id of a autoincremented type has been changed to `uint64` therefore we must modify this code to:
```
id := k.AppendPost(
    ctx,
    packet.SourcePort+"-"+packet.SourceChannel+"-"+data.Creator,
    data.Title,
    data.Content,
)
packetAck.PostID = strconv.FormatUint(id, 10)
```

And tells the user to add `"strconv"` in the imports just below `"errors"`

Reply:
I tested back the whole workflow, this is the only change to make :)

Reply:
We also need to mention to import `strconv`, if a dev doesn't use a proper editor.